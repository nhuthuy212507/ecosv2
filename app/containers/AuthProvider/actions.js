/*
 *
 * AuthProvider actions
 *
 */

import {
  GET_PROFILE,
  GET_PROFILE_SUCCESS,
  GET_PROFILE_ERROR,
} from './constants';

export function getProfile() {
  return {
    type: GET_PROFILE,
  };
}

export function getProfileSuccess(response) {
  return {
    type: GET_PROFILE_SUCCESS,
    response,
  };
}

export function getProfileError(error) {
  return {
    type: GET_PROFILE_ERROR,
    error,
  };
}
