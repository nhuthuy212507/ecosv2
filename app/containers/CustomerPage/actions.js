/*
 *
 * CustomerPage actions
 *
 */

import {
  GET_LIST_CUSTOMER,
  GET_LIST_CUSTOMER_SUCCESS,
  GET_LIST_CUSTOMER_ERROR,
  GET_OPTIONS_CUSTOMER_REFERENCE,
  GET_OPTIONS_CUSTOMER_REFERENCE_SUCCESS,
  GET_OPTIONS_CUSTOMER_REFERENCE_ERROR,
  GET_OPTIONS_CREATED_BY,
  GET_OPTIONS_CREATED_BY_SUCCESS,
  GET_OPTIONS_CREATED_BY_ERROR,
  ADD_CUSTOMER,
  ADD_CUSTOMER_SUCCESS,
  ADD_CUSTOMER_ERROR,
  EDIT_CUSTOMER,
  EDIT_CUSTOMER_SUCCESS,
  EDIT_CUSTOMER_ERROR,
  GET_CUSTOMER_EDITTING,
  GET_CUSTOMER_EDITTING_SUCCESS,
  GET_CUSTOMER_EDITTING_ERROR,
  GET_OPTIONS_CITY_WAREHOUSE_ADDRESS,
  GET_OPTIONS_CITY_WAREHOUSE_ADDRESS_SUCCESS,
  GET_OPTIONS_CITY_WAREHOUSE_ADDRESS_ERROR,
  GET_OPTIONS_DISTRICT_WAREHOUSE_ADDRESS,
  GET_OPTIONS_DISTRICT_WAREHOUSE_ADDRESS_SUCCESS,
  GET_OPTIONS_DISTRICT_WAREHOUSE_ADDRESS_ERROR,
  GET_OPTIONS_WARD_WAREHOUSE_ADDRESS,
  GET_OPTIONS_WARD_WAREHOUSE_ADDRESS_SUCCESS,
  GET_OPTIONS_WARD_WAREHOUSE_ADDRESS_ERROR,
  ADD_WAREHOUSE_ADDRESS,
  ADD_WAREHOUSE_ADDRESS_SUCCESS,
  ADD_WAREHOUSE_ADDRESS_ERROR,
  GET_LIST_WAREHOUSE_ADDRESS,
  GET_LIST_WAREHOUSE_ADDRESS_SUCCESS,
  GET_LIST_WAREHOUSE_ADDRESS_ERROR,
  EDIT_WAREHOUSE_ADDRESS,
  EDIT_WAREHOUSE_ADDRESS_SUCCESS,
  EDIT_WAREHOUSE_ADDRESS_ERROR,
} from './constants';

export function getListCustomer(params, actions = false) {
  return {
    type: GET_LIST_CUSTOMER,
    params,
    actions,
  };
}

export function getListCustomerSuccess(response) {
  return {
    type: GET_LIST_CUSTOMER_SUCCESS,
    response,
  };
}

export function getListCustomerError(error) {
  return {
    type: GET_LIST_CUSTOMER_ERROR,
    error,
  };
}

export function getOptionsCustomerReference(params) {
  return {
    type: GET_OPTIONS_CUSTOMER_REFERENCE,
    params,
  };
}

export function getOptionsCustomerReferenceSuccess(response) {
  return {
    type: GET_OPTIONS_CUSTOMER_REFERENCE_SUCCESS,
    response,
  };
}

export function getOptionsCustomerReferenceError(error) {
  return {
    type: GET_OPTIONS_CUSTOMER_REFERENCE_ERROR,
    error,
  };
}

export function getOptionsCreatedBy(params) {
  return {
    type: GET_OPTIONS_CREATED_BY,
    params,
  };
}

export function getOptionsCreatedBySuccess(response) {
  return {
    type: GET_OPTIONS_CREATED_BY_SUCCESS,
    response,
  };
}

export function getOptionsCreatedByError(error) {
  return {
    type: GET_OPTIONS_CREATED_BY_ERROR,
    error,
  };
}

export function addCustomer(data, actions = false) {
  return {
    type: ADD_CUSTOMER,
    data,
    actions,
  };
}

export function addCustomerSuccess() {
  return {
    type: ADD_CUSTOMER_SUCCESS,
  };
}

export function addCustomerError(error) {
  return {
    type: ADD_CUSTOMER_ERROR,
    error,
  };
}

export function getCustomerEditting(id) {
  return {
    type: GET_CUSTOMER_EDITTING,
    id,
  };
}

export function getCustomerEdittingSuccess(response) {
  return {
    type: GET_CUSTOMER_EDITTING_SUCCESS,
    response,
  };
}

export function getCustomerEdittingError(error) {
  return {
    type: GET_CUSTOMER_EDITTING_ERROR,
    error,
  };
}

export function editCustomer(id, data, actions = false) {
  return {
    type: EDIT_CUSTOMER,
    id,
    data,
    actions,
  };
}

export function editCustomerSuccess() {
  return {
    type: EDIT_CUSTOMER_SUCCESS,
  };
}

export function editCustomerError(error) {
  return {
    type: EDIT_CUSTOMER_ERROR,
    error,
  };
}

export function getOptionsCityWarehouseAddress(params) {
  return {
    type: GET_OPTIONS_CITY_WAREHOUSE_ADDRESS,
    params,
  };
}

export function getOptionsCityWarehouseAddressSuccess(response) {
  return {
    type: GET_OPTIONS_CITY_WAREHOUSE_ADDRESS_SUCCESS,
    response,
  };
}

export function getOptionsCityWarehouseAddressError(error) {
  return {
    type: GET_OPTIONS_CITY_WAREHOUSE_ADDRESS_ERROR,
    error,
  };
}

export function getOptionsDistrictWarehouseAddress(params) {
  return {
    type: GET_OPTIONS_DISTRICT_WAREHOUSE_ADDRESS,
    params,
  };
}

export function getOptionsDistrictWarehouseAddressSuccess(response) {
  return {
    type: GET_OPTIONS_DISTRICT_WAREHOUSE_ADDRESS_SUCCESS,
    response,
  };
}

export function getOptionsDistrictWarehouseAddressError(error) {
  return {
    type: GET_OPTIONS_DISTRICT_WAREHOUSE_ADDRESS_ERROR,
    error,
  };
}

export function getOptionsWardWarehouseAddress(params) {
  return {
    type: GET_OPTIONS_WARD_WAREHOUSE_ADDRESS,
    params,
  };
}

export function getOptionsWardWarehouseAddressSuccess(response) {
  return {
    type: GET_OPTIONS_WARD_WAREHOUSE_ADDRESS_SUCCESS,
    response,
  };
}

export function getOptionsWardWarehouseAddressError(error) {
  return {
    type: GET_OPTIONS_WARD_WAREHOUSE_ADDRESS_ERROR,
    error,
  };
}

export function addWarehouseAddress(data, actions = false) {
  return {
    type: ADD_WAREHOUSE_ADDRESS,
    data,
    actions,
  };
}

export function addWarehouseAddressSuccess() {
  return {
    type: ADD_WAREHOUSE_ADDRESS_SUCCESS,
  };
}

export function addWarehouseAddressError(error) {
  return {
    type: ADD_WAREHOUSE_ADDRESS_ERROR,
    error,
  };
}

export function getListWarehouseAddress(params, actions = false) {
  return {
    type: GET_LIST_WAREHOUSE_ADDRESS,
    params,
    actions,
  };
}

export function getListWarehouseAddressSuccess(response) {
  return {
    type: GET_LIST_WAREHOUSE_ADDRESS_SUCCESS,
    response,
  };
}

export function getListWarehouseAddressError(error) {
  return {
    type: GET_LIST_WAREHOUSE_ADDRESS_ERROR,
    error,
  };
}

export function editWarehouseAddress(data, actions = false) {
  return {
    type: EDIT_WAREHOUSE_ADDRESS,
    data,
    actions,
  };
}

export function editWarehouseAddressSuccess() {
  return {
    type: EDIT_WAREHOUSE_ADDRESS_SUCCESS,
  };
}

export function editWarehouseAddressError(error) {
  return {
    type: EDIT_WAREHOUSE_ADDRESS_ERROR,
    error,
  };
}

export function deleteWarehouseAddress(data, actions = false) {
  return {
    type: EDIT_WAREHOUSE_ADDRESS,
    data,
    actions,
  };
}

export function deleteWarehouseAddressSuccess() {
  return {
    type: EDIT_WAREHOUSE_ADDRESS_SUCCESS,
  };
}

export function deleteWarehouseAddressError(error) {
  return {
    type: EDIT_WAREHOUSE_ADDRESS_ERROR,
    error,
  };
}
