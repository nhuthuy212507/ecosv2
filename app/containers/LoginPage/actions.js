/*
 *
 * LoginPage actions
 *
 */

import { LOGIN, LOGIN_SUCCESS, LOGIN_ERROR } from './constants';

export function login(user) {
  return {
    type: LOGIN,
    user,
  };
}

export function loginSuccess(token) {
  return {
    type: LOGIN_SUCCESS,
    token,
  };
}

export function loginError(error) {
  return {
    type: LOGIN_ERROR,
    error,
  };
}
