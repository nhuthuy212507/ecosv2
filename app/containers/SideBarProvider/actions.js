/*
 *
 * SideBarProvider actions
 *
 */

import {
  CHANGE_DESKTOP_SIDEBAR_VISIBILITY,
  CHANGE_MOBILE_SIDEBAR_VISIBILITY,
} from './constants';

export function changeDesktopSidebarVisibility() {
  return {
    type: CHANGE_DESKTOP_SIDEBAR_VISIBILITY,
  };
}

export function changeMobileSidebarVisibility() {
  return {
    type: CHANGE_MOBILE_SIDEBAR_VISIBILITY,
  };
}
