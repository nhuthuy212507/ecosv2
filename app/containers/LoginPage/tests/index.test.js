/**
 * Test the HomePage
 */

import React from 'react';
import { shallow } from 'enzyme';
import { IntlProvider } from 'react-intl';

import { LoginPage, mapDispatchToProps } from '../index';
import { login } from '../actions';

describe('<LoginPage />', () => {
  it('should render and match the snapshot', () => {
    const loginSpy = jest.fn();
    const wrapper = shallow(
      <IntlProvider locale="en">
        <LoginPage onLogin={loginSpy} />
      </IntlProvider>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  describe('mapDispatchToProps', () => {
    describe('onLogin', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.onLogin).toBeDefined();
      });

      it('should dispatch onLogin when called', () => {
        const user = {
          username: 'test',
          password: 'test',
        };
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        result.onLogin(user);
        expect(dispatch).toHaveBeenCalledWith(login(user));
      });
    });
  });
});
