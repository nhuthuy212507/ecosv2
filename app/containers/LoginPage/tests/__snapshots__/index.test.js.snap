// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginPage /> should render and match the snapshot 1`] = `ShallowWrapper {}`;
